### DISCUSSION TITLE
Read() - What if the characteristic id is a JSON object?

### DISCUSSION THREAD

**Discussion by dustinHahn** [capacitor-community/bluetooth-le#694] [Q&A]
I am can connect to a BLE device and need to be able to read/write the characteristics using a UAR_service & URT_TX_SERVICE. 

UART_SVC= '6E40001-##############'
UART_TX_SVC = '6E400003-###########'
DEV_POWER = textToDataView('{"deviceSetup":{"power":""}}');

getPowerProp (){
     await BleClient.write(deviceId, this.UART_SVC, this.UART.TX_SVC, this.DEV_POWER)
     console.log('DEV_POWER', this.DEV_POWER)
}
 **the value logged out is ""

**Comment by peitschie** [comment]
Hi @dustinHahn 

Bluetooth is unrelated to your problem here.

What you are demonstrating here is completely about what the browser logs when handed a `DataView` object.

E.g., this will give the same result (note: there's no call to Bluetooth at all!):
```
DEV_POWER = textToDataView('{"deviceSetup":{"power":""}}');
console.log('DEV_POWER', DEV_POWER);
```

Some browsers do not do a good job rendering a data class like this. You can prove this out by using `dataViewToText` to convert the data view back. E.g., see what this gives you:
```
DEV_POWER = textToDataView('{"deviceSetup":{"power":""}}');
DEV_POWER_AS_STRING = dataViewToText(DEV_POWER);
console.log('DEV_POWER', DEV_POWER_AS_STRING );
```
